diff -ur migemo-0.40-orig/configure migemo-0.40/configure
--- migemo-0.40-orig/configure	2003-05-28 12:19:38.000000000 +0000
+++ migemo-0.40/configure	2020-02-11 04:37:32.040348859 +0000
@@ -968,7 +968,7 @@
 echo "configure:969: checking where .rb files should go" >&5
   if test "x$rubydir" = x; then
     
-    rubydir=`ruby -rrbconfig -e 'puts Config::CONFIG["sitedir"]'`
+    rubydir=`ruby -rrbconfig -e 'puts RbConfig::CONFIG["sitelibdir"]'`
     
   fi
   echo "$ac_t""$rubydir" 1>&6
diff -ur migemo-0.40-orig/genchars.sh migemo-0.40/genchars.sh
--- migemo-0.40-orig/genchars.sh	2001-08-13 09:30:48.000000000 +0000
+++ migemo-0.40/genchars.sh	2020-02-11 04:37:32.040348859 +0000
@@ -1,6 +1,7 @@
 #! /bin/sh
+RUBY=${RUBY:-ruby}
 
-ruby -rromkan -nle 'head = split[0]; if /^\w+$/ =~ head then puts head else roma = head.to_roma; puts roma, roma.to_kunrei end' migemo-dict |uniq> tmp.ascii.words
+${RUBY} -rromkan -nle '$_.force_encoding("EUC-JP") if defined?(Encoding); head = $_.split[0]; if /^\w+$/ =~ head then puts head else roma = head.to_roma; puts roma, roma.to_kunrei end' migemo-dict |uniq> tmp.ascii.words
 
 # Get the top 500 frequent ngrams.
 for i in 1 2 3 4 5 6 7 8; do
diff -ur migemo-0.40-orig/migemo migemo-0.40/migemo
--- migemo-0.40-orig/migemo	2003-05-27 03:01:10.000000000 +0000
+++ migemo-0.40/migemo	2020-02-11 04:37:32.040348859 +0000
@@ -10,7 +10,7 @@
 # the GNU General Public License version 2.
 #
 
-$KCODE = "e"
+$KCODE = 'e' unless defined?(Encoding)
 
 require 'migemo'
 require 'getoptlong'
@@ -140,6 +140,9 @@
   }
 
   $stdout.sync = true
+  if defined?(Encoding)
+    $stdin.set_encoding('EUC-JP')
+  end
   while line = gets
     pattern = line.chomp
     segments = segment(pattern)
diff -ur migemo-0.40-orig/migemo-cache.rb migemo-0.40/migemo-cache.rb
--- migemo-0.40-orig/migemo-cache.rb	2001-07-14 17:38:56.000000000 +0000
+++ migemo-0.40/migemo-cache.rb	2020-02-11 04:37:32.040348859 +0000
@@ -1,5 +1,5 @@
 require 'migemo'
-$KCODE="e"
+$KCODE = 'e' unless defined?(Encoding)
 raise if ARGV[0] == nil
 dict = ARGV[0]
 static_dict = MigemoStaticDict.new(dict)
@@ -18,10 +18,15 @@
   migemo = Migemo.new(static_dict, pattern)
   migemo.optimization = 3
   data = Marshal.dump(migemo.regex_tree)
-  output = [pattern.length].pack("N") + pattern + 
-    [data.length].pack("N") + data
+  if defined?(Encoding) then
+    output = [pattern.bytesize].pack('N') + pattern.encode('ASCII-8BIT') +
+      [data.bytesize].pack('N') + data
+  else
+    output = [pattern.length].pack('N') + pattern +
+      [data.length].pack('N') + data
+  end
   cache.print output
   index.print [idx].pack("N")
-  idx += output.length
+  idx += output.bytesize
 end
 
diff -ur migemo-0.40-orig/migemo-client migemo-0.40/migemo-client
--- migemo-0.40-orig/migemo-client	2001-08-13 09:30:48.000000000 +0000
+++ migemo-0.40/migemo-client	2020-02-11 04:37:32.041348862 +0000
@@ -10,7 +10,7 @@
 # the GNU General Public License version 2.
 #
 
-$KCODE = "e"
+$KCODE = 'e' unless defined?(Encoding)
 require 'net/http'
 require 'getoptlong'
 
diff -ur migemo-0.40-orig/migemo-convert.rb migemo-0.40/migemo-convert.rb
--- migemo-0.40-orig/migemo-convert.rb	2003-05-26 06:55:22.000000000 +0000
+++ migemo-0.40/migemo-convert.rb	2020-02-11 04:37:32.041348862 +0000
@@ -1,3 +1,4 @@
+# -*- coding: euc-jp -*-
 #
 # Ruby/Migemo - a library for Japanese incremental search.
 #
@@ -14,7 +15,8 @@
 #
 # Convert a SKK's dictionary into Migemo's.
 #
-$KCODE= "e"
+$KCODE = 'e' unless defined?(Encoding)
+
 require 'romkan'
 
 HIRAGANA = "[¤¡-¤ó¡¼¡Á]"
@@ -25,6 +27,7 @@
 puts ";;"
 lines = readlines
 while line = lines.shift
+  line.force_encoding('EUC-JP') if defined?(Encoding)
   if /^;/ =~ line
     puts line
   else
@@ -36,6 +39,7 @@
 
 dict = [];
 while line = lines.shift
+  line.force_encoding('EUC-JP') if defined?(Encoding)
   if /^(#{HIRAGANA}+)[a-z]? (.*)/ =~ line || /^(\w+) (.*)/ =~ line 
     head = $1
     words = $2.split('/').map {|x| 
diff -ur migemo-0.40-orig/migemo-dict.rb migemo-0.40/migemo-dict.rb
--- migemo-0.40-orig/migemo-dict.rb	2002-10-22 05:38:14.000000000 +0000
+++ migemo-0.40/migemo-dict.rb	2020-02-11 04:37:32.041348862 +0000
@@ -34,6 +34,8 @@
 class MigemoDict
   def initialize (filename)
     @dict  = File.new(filename)
+    @dict.set_encoding('EUC-JP') if defined?(Encoding)
+    return @dict
   end
 
   private
@@ -106,6 +108,7 @@
 class MigemoDictCache
   def initialize (filename)
     @dict  = File.new(filename)
+    @dict.set_encoding('EUC-JP') if defined?(Encoding)
     @index = File.new(filename + ".idx").read.unpack "N*"
   end
 
diff -ur migemo-0.40-orig/migemo-grep migemo-0.40/migemo-grep
--- migemo-0.40-orig/migemo-grep	2003-05-27 07:43:07.000000000 +0000
+++ migemo-0.40/migemo-grep	2020-02-11 04:37:32.041348862 +0000
@@ -9,7 +9,7 @@
 # You can redistribute it and/or modify it under the terms of 
 # the GNU General Public License version 2.
 #
-$KCODE = "e"
+$KCODE = 'e' unless defined?(Encoding)
 
 require 'migemo'
 require 'getoptlong'
diff -ur migemo-0.40-orig/migemo-index.rb migemo-0.40/migemo-index.rb
--- migemo-0.40-orig/migemo-index.rb	2003-05-26 06:45:53.000000000 +0000
+++ migemo-0.40/migemo-index.rb	2020-02-11 04:37:32.041348862 +0000
@@ -16,8 +16,9 @@
 #
 offset = 0
 while line = gets
+  line.force_encoding('EUC-JP') if defined?(Encoding)
   unless line =~ /^;/
     print [offset].pack("N")
   end
-  offset += line.length
+  offset += line.bytesize
 end
diff -ur migemo-0.40-orig/migemo.rb.in migemo-0.40/migemo.rb.in
--- migemo-0.40-orig/migemo.rb.in	2003-05-28 12:00:52.000000000 +0000
+++ migemo-0.40/migemo.rb.in	2020-02-11 04:37:32.042348865 +0000
@@ -1,3 +1,4 @@
+# -*- coding: euc-jp -*-
 #
 # Ruby/Migemo - a library for Japanese incremental search.
 #
@@ -14,7 +15,7 @@
 require 'migemo-dict'
 require 'migemo-regex'
 require 'romkan'
-require 'jcode'
+require 'jcode' unless defined?(Encoding)
 include MigemoRegex
 
 class String
@@ -24,7 +25,7 @@
   end
 
   def quotemeta
-    self.gsub(/([^ \w])/, '\\\\\\1')
+    self.gsub(/([^ \p{Word}])/, '\\\\\\1')
   end
 
   def first
diff -ur migemo-0.40-orig/tests/Makefile.in migemo-0.40/tests/Makefile.in
--- migemo-0.40-orig/tests/Makefile.in	2003-05-29 08:09:03.000000000 +0000
+++ migemo-0.40/tests/Makefile.in	2020-02-11 04:37:32.042348865 +0000
@@ -201,10 +201,11 @@
 	ruby -I.. ../migemo-index.rb test-dict > test-dict.idx
 
 test-dict.cache: test-dict test-dict.idx ../migemo-cache.rb
-	ruby -rromkan -ne 'puts $$1.to_roma if /^(.+?)	/' test-dict |\
-	while read line; do\
-	    echo $$line | ruby -ne 'chomp!;1.upto($$_.length) do |x| puts $$_[0,x] end';\
-	done | ruby -I.. ../migemo-cache.rb test-dict
+	$(RUBY) -rromkan -ne '$$_.force_encoding("EUC-JP") if defined?("".encoding); puts $$1.to_roma if /^(.+?)	/' test-dict > tmp.list1
+	cat tmp.list1 | while read line; do\
+	    echo $$line | $(RUBY) -ne '$$_.chomp!;1.upto($$_.length) do |x| puts $$_[0,x] end';\
+	done > tmp.chars
+	cat tmp.chars | $(RUBY) -I.. ../migemo-cache.rb test-dict
 
 clean-local:
 	rm -f tmp.*
